components/sup/src/manager/service/hooks.rs

```diff
@@ -173,9 +173,7 @@ impl Hook for RunHook {
                   stderr_log_path: hooks::stderr_log_path::<Self>(package_name), }
     }
 
-    fn run<T>(&self, _: &str, _: &Pkg, _: Option<T>) -> Self::ExitValue
-        where T: ToString
-    {
+    fn run(&self, _: &str, _: &Pkg, _: Option<String>) -> Self::ExitValue {
```

Comment:
Why change this when it requires the addition of `clone`s in the callers? I think you could change it to `Option<&str>`, and then the callers could do `.map(|s| s.as_str())` on the `Option<String>` values they own.

Reply:
If you follow the calls all the way through to the end, `.to_string()` ends up getting called anyway, so the allocation happens regardless.  All of this came from one of the generic parameters in `ServiceCredential::new` being passed by value instead of ref, and pulling on that string unraveled to this point.  I can change it to `Option<&str>` though.

Reply:
Fair enough! Carry on: either way is fine with me, so whichever you like is 💯

Reply:
It turns out that converting this to `Option<&str>` is less straightforward than I had hoped, but I understand that the presence of so many `clone()`s is also sub-optimal.  I think in the interest of time, I'm going to back all of this out back to the way it was before I made any changes, and just allow that single clippy lint on `ServiceCredential`, with a TODO to fix it later, along with the rest.

Reply:
That approach sounds good. In lieu of a `TODO` (which I'm really not a fan of)`, can we re-open https://github.com/habitat-sh/habitat/issues/6213 and add it to the list of work?

Reply:
Yep, sounds good.